packages/core-magistrate-transactions/src/handlers/bridgechain-registration.ts

```diff
@@ -65,13 +70,20 @@ export class BridgechainRegistrationTransactionHandler extends MagistrateTransac
         const { data }: Interfaces.ITransaction = transaction;
         const bridgechains: Record<string, IBridgechainWalletAttributes> = wallet.getAttribute("business.bridgechains");
 
-        if (
-            bridgechains &&
-            Object.values(bridgechains).some(bridgechain => {
-                return bridgechain.bridgechainAsset.name === data.asset.bridgechainRegistration.name;
-            })
-        ) {
-            throw new BridgechainAlreadyRegisteredError();
+        if (bridgechains) {
+            const bridgechainValues: IBridgechainWalletAttributes[] = Object.values(bridgechains);
+
+            for (let i = 0, len = bridgechainValues.length; i < len; i++) {
```

Comment:
```suggestion
            for (const bridgechain of bridgechainValues) {
```

Easier to read and there will never be an amount of bridgechains or businesses to iterate over where a raw `for` loop would have any significant benefits, especially with the v8 optimisations for `for of` loops.